### PR TITLE
Make test tolerance dependent on condition number in Bunch-Kaufman test

### LINKED ISSUE #25096: Julia master: Error in testset bunchkaufman: SSL certificate is invalid
In:
```
julia> versioninfo()
Julia Version 0.7.0-DEV.2976
Commit 1f33cddf9d (2017-12-14 23:41 UTC)
Platform Info:
  OS: Linux (x86_64-suse-linux)
  CPU: Intel(R) Core(TM) i5-4460  CPU @ 3.20GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, haswell)
Environment:
```
make testall fails with:
```
Error in testset linalg/bunchkaufman:
Test Failed at /home/colin/Downloads/julia7/test/linalg/bunchkaufman.jl:98
  Expression: ≈(apd * (bc2 \ b), b, atol=150000ε)
   Evaluated: [2.0 1.0; 5.0 4.0; … ; 3.0 5.0; 2.0 5.0] ≈ [2 1; 5 4; … ; 3 5; 2 5] (atol=3.3306690738754696e-11) 
Error in testset linalg/bunchkaufman:
Test Failed at /home/colin/Downloads/julia7/test/linalg/bunchkaufman.jl:98
  Expression: ≈(apd * (bc2 \ b), b, atol=150000ε)
   Evaluated: [2.0 1.0; 5.0 4.0; … ; 3.0 5.0; 2.0 5.0] ≈ [2 1; 5 4; … ; 3 5; 2 5] (atol=3.3306690738754696e-11) 
Error in testset linalg/bunchkaufman:
Test Failed at /home/colin/Downloads/julia7/test/linalg/bunchkaufman.jl:98
  Expression: ≈(apd * (bc2 \ b), b, atol=150000ε)
   Evaluated: [2.0 1.0; 5.0 4.0; … ; 3.0 5.0; 2.0 5.0] ≈ [2 1; 5 4; … ; 3 5; 2 5] (atol=3.3306690738754696e-11) 
Error in testset linalg/bunchkaufman:
Test Failed at /home/colin/Downloads/julia7/test/linalg/bunchkaufman.jl:98
  Expression: ≈(apd * (bc2 \ b), b, atol=150000ε)
   Evaluated: [2.0 1.0; 5.0 4.0; … ; 3.0 5.0; 2.0 5.0] ≈ [2 1; 5 4; … ; 3 5; 2 5] (atol=3.3306690738754696e-11) 
Error in testset libgit2:
Test Failed at /home/colin/Downloads/julia7/test/libgit2.jl:2624
  Expression: err.code == LibGit2.Error.ECERTIFICATE
   Evaluated: ERROR::Base.LibGit2.Error.Code = -1 == ECERTIFICATE::Base.LibGit2.Error.Code = -17
Error in testset libgit2:
Test Failed at /home/colin/Downloads/julia7/test/libgit2.jl:2625
  Expression: startswith(lowercase(err.msg), lowercase("The SSL certificate is invalid"))
Error in testset libgit2:
Test Failed at /home/colin/Downloads/julia7/test/libgit2.jl:2638
  Expression: lowercase(err.msg) == lowercase("invalid Content-Type: text/plain")
   Evaluated: "curl error: failed to connect to localhost port 49152: connection refused\n" == "invalid content-type: text/plain"
Error in testset libgit2:
Test Failed at /home/colin/Downloads/julia7/test/libgit2.jl:2642
  Expression: process_running(pobj)
ERROR: LoadError: Test run finished with errors
in expression starting at /home/colin/Downloads/julia7/test/runtests.jl:46
Makefile:20: recipe for target 'all' failed
make[1]: *** [all] Error 1
Makefile:540: recipe for target 'testall' failed
make: *** [testall] Error 2
```
Gist at https://gist.github.com/colbec/2b3cd4dffd30dd16770f2eea6f84d82e

### PR DESCRIPTION
**Pull request by andreasnoack**
Fixes #25096

### CODE REVIEW

**Comment by rfourquet** [comment]
If this can help, this error can be seen [on CI](https://ci.appveyor.com/project/JuliaLang/julia/build/1.0.22286/job/l097utwfwh6m26gd), with RNG seed `0x48c527989c99cb3db9f41553cee8fc9e`.

File: test/linalg/bunchkaufman.jl [hunk 1]
```diff
@@ -19,7 +19,7 @@ a2img  = randn(n,n)/2
 breal = randn(n,2)/2
 bimg  = randn(n,2)/2
 
-@testset "$eltya argument A" for eltya in (Float32, Float64, ComplexF32, ComplexF64, Int)
+@testset "$eltya argument A" for eltya in (Float32, Float64, Complex64, Complex128, Int)
```
Comment (rfourquet) [inline test/linalg/bunchkaufman.jl]: Slowly reverting the `Complex` renaming PR? :wink:

Comment (andreasnoack) [reply test/linalg/bunchkaufman.jl]: Oops.

**Review by Sacha0 — APPROVED** [review]
lgtm modulo AV i686 signoff! Thanks `@andreasnoack`! :)

**Pull request merged**